extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaRyaStreamsClientFactory.java

```
@@ -0,0 +1,169 @@
+
+        // Setup Query Repository used by the Kafka Rya Streams subsystem.
+        final Producer<?, QueryChange> queryProducer =
+                makeProducer(kafkaHostname, kafkaPort, StringSerializer.class, QueryChangeSerializer.class);
+        final Consumer<?, QueryChange>queryConsumer =
+                fromStartConsumer(kafkaHostname, kafkaPort, StringDeserializer.class, QueryChangeDeserializer.class);
+        final String changeLogTopic = KafkaTopics.queryChangeLogTopic(ryaInstance);
+        final QueryChangeLog changeLog = new KafkaQueryChangeLog(queryProducer, queryConsumer, changeLogTopic);
+        final QueryRepository queryRepo = new InMemoryQueryRepository(changeLog);
+
+        // Create the Rya Streams client that is backed by a Kafka Query Change Log.
+        return new RyaStreamsClient(
+                new DefaultAddQuery(queryRepo),
+                new DefaultGetQuery(queryRepo),
+                new DefaultDeleteQuery(queryRepo),
+                new KafkaGetQueryResultStream<>(kafkaHostname, "" + kafkaPort, VisibilityStatementDeserializer.class),
+                new KafkaGetQueryResultStream<>(kafkaHostname, "" + kafkaPort, VisibilityBindingSetDeserializer.class),
+                new DefaultListQueries(queryRepo),
+                new DefaultStartQuery(queryRepo),
+                new DefaultStopQuery(queryRepo)) {
+
+            /**
+             * Close the QueryRepository used by the returned client.
+             */
+            @Override
+            public void close() {
+                try {
+                    queryRepo.close();
+                } catch (final Exception e) {
+                    log.warn("Couldn't close a QueryRepository.", e);
+                }
+            }
+        };
+    }
+
+    /**
+     * Create a {@link Producer} that is able to write to a topic in Kafka.
+     *
+     * @param kafkaHostname - The Kafka broker hostname. (not null)
+     * @param kafkaPort - The Kafka broker port.
```

port restrictions?

---

I'd rather just let it throw an exception.

---

add a doc saying as much?

---

I don't think it's needed.